src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -329,7 +334,9 @@ def autoregressive_inference(
         engine_inputs.update(kv_cache)
         engine_inputs = [engine_inputs[name] for name in self.engine.input_names]
 
-        new_logits, *cache_values = self.engine(engine_inputs)
+        new_logits, *cache_values = self.engine(engine_inputs, kv_cache=kv_cache)
+        self.kv_cache.shift_last(1)
+        # TODO: do we still need to update the cache below since engine handles?
```

Comment:
If 'shift last' will work correctly also in the case of onnx runtime inference, it would be great! we could get rid of assemble_kv_cache  I think.

Reply:
FWIW, the KVCache object ONLY works with the deepsparse engine. Onnx runtime will not be able to leverage this infrastructure.